### PR TITLE
Update update powerfox2 to 0.0.13

### PR DESCRIPTION
**Pull request by Ax-LED**
updated powerfox2 from version 12 to 13

Regards

AxLED

### CODE REVIEW

**Comment by mcm1957** [comment]
0.0.13 created 16.11.2023

**Comment by Ax-LED** [comment]
@mcm1957 
as i am not publishing new adapter versions every day, i need some help.

1. Do you know, how i can fix these errors/warnings:
❗ [E405] Icon (latest) must be in the following path: https://raw.githubusercontent.com/Ax-LED/ioBroker.powerfox2/main/
 ❗ [E407] Meta URL (latest) must be equal to https://raw.githubusercontent.com/Ax-LED/ioBroker.powerfox2/main/io-package.json
 ❗ [E426] Icon (stable) must be in the following path: https://raw.githubusercontent.com/Ax-LED/ioBroker.powerfox2/main/
 ❗ [E428] Meta URL (stable) must be equal to https://raw.githubusercontent.com/Ax-LED/ioBroker.powerfox2/main/io-package.json
 ❗ [E518] "@alcalzone/release-script" is used, but ".releaseconfig.json" not found

2. after editing the files, how can i update the pull request?

Looking forward hearing from you.

Best regards

AxLED

**Comment by mcm1957** [comment]
No problem :-)

The errors 
```
❗ [E405] Icon (latest) must be in the following path: https://raw.githubusercontent.com/Ax-LED/ioBroker.powerfox2/main/
❗ [E407] Meta URL (latest) must be equal to https://raw.githubusercontent.com/Ax-LED/ioBroker.powerfox2/main/io-package.json
❗ [E426] Icon (stable) must be in the following path: https://raw.githubusercontent.com/Ax-LED/ioBroker.powerfox2/main/
❗ [E428] Meta URL (stable) must be equal to https://raw.githubusercontent.com/Ax-LED/ioBroker.powerfox2/main/io-package.json
```
are caused by moving from master to main and must be corrected at the repository json files. I'm already fixing this.

I'll write a summary later what to do by you. Please wait a little bit.

**Comment by mcm1957** [comment]
RE-CHECK!

**Comment by mcm1957** [comment]
@Ax-LED 

Here is me feedback:

- [ ]  ❗ [E518] "@alcalzone/release-script" is used, but ".releaseconfig.json" not found

As you are still using @alcalzone/releasescritp 2.x.x you can ignore this error. But remember that .releaseconfig is required for @alcalzone/releasescript 3.x.x. See README of releasescript

- [ ] ❗ [E605] No actual year found in copyright. Please add "Copyright (c) 2023 Ax-LED [alex_draxler@yahoo.de](mailto:alex_draxler@yahoo.de)" at the end of [README.md](https://github.com/Ax-LED/ioBroker.powerfox2/blob/master/README.md)
  ❗ [E701] No actual year found in LICENSE. Please add "Copyright (c) 2023 Ax-LED [alex_draxler@yahoo.de](mailto:alex_draxler@yahoo.de)" at the start of LICENSE

Please add copyright lines at README and LICENSE

- [ ] The following warnings can be ignored for now. Missing uk language will be added the next time you use npm run translat (with an up to date versoion). Those warnings rae NOT blocking stable:

 👀 [W171] "common.title" is deprecated in [io-package.json](https://github.com/Ax-LED/ioBroker.powerfox2/blob/master/io-package.json)
 👀 [W105] "common.titleLang" should be translated into all supported languages (en, de, ru, pt, nl, fr, it, es, pl, uk, zh-cn)
 👀 [W109] "common.desc" should be translated into all supported languages (en, de, ru, pt, nl, fr, it, es, pl, uk, zh-cn)
 👀 [W156] Adapter should support admin 5 UI (jsonConfig) if you do not use a React based UI
 👀 [W145] Each "common.news" should be translated into all supported languages (en, de, ru, pt, nl, fr, it, es, pl, uk, zh-cn)

- [ ]  👀 [W513] "gulpfile.js" found in repo! Think about migrating to @iobroker/adapter-dev package

If you do not use gulp you might consider removing this file and gulp dependy.

- [ ] Testing must be updated to test node 16 / 18 and 20. 
  Please update test-and-release.yml and trigger a run by i.e. performing any commit. A successful tun is required fpor release to stable. If you need halp, I can provide a PR with te release adaption.

- [ ] As you are using alcalcones release script you might consider to enable automatic deploy to npm. Simply uncomment the lines at test-and-release and store npm token at github secrets as described in test-and-release.yml. But of yourse this is up to you, If you prefer to do deploy manually its ok too, aumatis should / could only help to reduce manual steps. But this is nogt required for stable release.

To summarize: All issues above doe nOT need a new release, simply fixing and triggering a test-and-release workflow action (whoich succeeds) is sufficient.

**Please drop acomment if you have ana question, need help or think that the above problems are fixed.**

**Comment by Ax-LED** [comment]
@mcm1957 

1. I was (hopefully) able to change the code to fix the following errors:
❗ [E605] No actual year found in copyright. Please add "Copyright (c) 2023 Ax-LED [alex_draxler@yahoo.de](mailto:alex_draxler@yahoo.de)" at the end of [README.md](https://github.com/Ax-LED/ioBroker.powerfox2/blob/master/README.md)
❗ [E701] No actual year found in LICENSE. Please add "Copyright (c) 2023 Ax-LED [alex_draxler@yahoo.de](mailto:alex_draxler@yahoo.de)" at the start of LICENSE

2. I did a new github upload. How can i check if errors from 1. are fixed?
3. I don't how i installed/integrated alcalzones script in the past, so i left it how it was
4. How can i find out, if i use/need gulpfile.js?
5. Question to: Testing must be updated to test node 16 / 18 and 20. I am not able to find test-and-release.yml

AxLED

**Comment by mcm1957** [comment]
To retrigger the check add a commnte RE-CHECK!
image

**Comment by mcm1957** [comment]
RE-CHECK!

**Comment by mcm1957** [comment]
@Ax-LED 
OK - copyright problems ere fixed.

I'll create a PR tomorrow to update test-and-release.yml. This file is a github workflo and located at .github/workflows.
I will activate deployment too as you are using releasescript anyway.

**Comment by ticaki** [comment]
To help mcm a bit:
I had a look at the dependencies in the package.json and axios should be under dependencies and not devDependencies fix with:
Commands in iobroker.powerfox2/
```
npm remove -D axios
npm install axios
```

The following is displayed as an error in vscode:

```
this.config.email
```
email is missing in the io-package.json under `native:{}` 

and the line 8 in .vscode/settings.json should be exchanged for this:
```
"url": "https://raw.githubusercontent.com/ioBroker/ioBroker.js-controller/master/schemas/io-package.json"
```

**Comment by mcm1957** [comment]
> To help mcm a bit: I had a look at the dependencies in the package.json
Thanks

> and axios should be under dependencies and not devDependencies fix with: Commands in iobroker.powerfox2/
I'll provide a PR and explain a little bit at forum. This change might require some testing. For the moment I'll accept the incorrect dependency as it exists since years and did not (yet) harm. But it must be fixed for next release for sure.

> email is missing in the io-package.json under `native:{}`
Thats correct but minor and not blocking for stable. I'll explain details at forum

> and the line 8 in .vscode/settings.json should be exchanged for this:
> 
> ```
> "url": "https://raw.githubusercontent.com/ioBroker/ioBroker.js-controller/master/schemas/io-package.json"
> ```
That relates to you vscode usage only. It dies not effect stable release

**Comment by mcm1957** [comment]
PRs are available
Please merge whenever you have time
As tests are OK at forked repo nothing is blocking stable release anymore except the minimum time at latest.

**Comment by Ax-LED** [comment]
@ticaki 

i fixed your hints about
- axios
- settings.json

i didnt touch io-package.json so far, as i don't know if the adapter is working anymore if i change it.

AxLED

**Comment by ticaki** [comment]
These are merge conflicts, the update of Axios has changed your package.json and therefore git cannot merge them without help.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github

 If your conflict is too complex for Github and you need to continue on the command line, please ask again.


other thing: its near impossible that the native change of io-package.json break ur adapter.

**Comment by Ax-LED** [comment]
@ticaki 
@mcm1957 
i corrected your suggestions:
email is missing in the io-package.json under native:{}
Thats correct but minor and not blocking for stable. I'll explain details at forum

AxLED

**Comment by Ax-LED** [comment]
had to withdraw the PR, had different versions on npm and github. After testing version 0.0.14 by forum users, I will provide a new PR.

AxLED
